### PR TITLE
elb Unhealthy Instances

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -1403,4 +1404,4 @@ module.exports = {
         'securityAgentInstalled'        : require(__dirname + '/plugins/alibaba/securitycenter/securityAgentInstalled.js'),
         'securityNotificationsEnabled'  : require(__dirname + '/plugins/alibaba/securitycenter/securityNotificationsEnabled.js')
     }
-};
+};
```
Comment (fatima99s) [inline exports.js:1407]:
```suggestion
};

```

File: plugins/aws/elb/elbUnhealthyInstances.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'ELB Unhealthy Instances',
+    category: 'ELB',
+    domain: 'Content Delivery',
+    description: 'Detects ELBs that have unhealthy instances attached',
```
Comment (alphadev4) [inline plugins/aws/elb/elbUnhealthyInstances.js:8]:
```suggestion
    description: 'Ensures that AWS ELBs have healthy instances attached',
```

File: plugins/aws/elb/elbUnhealthyInstances.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    more_info: 'ELBs should have healthy instances attached to ensure proper load balancing and availability. Unhealthy instances can indicate issues with the backend server resources.',
```
Comment (alphadev4) [inline plugins/aws/elb/elbUnhealthyInstances.js:9]:
```suggestion
    more_info: 'ELBs should have healthy instances attached to ensure proper load balancing and availability. The status of the instances that are healthy should be InService.',
```

File: plugins/aws/elb/elbUnhealthyInstances.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    link: 'https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-healthchecks.html',
```
Comment (alphadev4) [inline plugins/aws/elb/elbUnhealthyInstances.js:10]:
```suggestion
    link: 'https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-healthchecks.html#check-instance-health',
```

**Review by alphadev4 — CHANGES_REQUESTED** [review]

File: plugins/aws/elb/elbUnhealthyInstances.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                var instanceStates = describeInstanceHealth.data.InstanceStates;
+            
+                var unhealthyInstances = instanceStates.filter(function(instance) {
+                    return instance.State === 'OutOfService';
+                });
+            
+                if (unhealthyInstances.length > 0) {
+                    helpers.addResult(results, 2, `ELB has ${unhealthyInstances} unhealthy instance(s)`, region, elbArn);
+                } else {
+                    helpers.addResult(results, 0, `ELB does not have unhealthy instance`, region, elbArn);
```
Comment (mehakseedat63) [inline plugins/aws/elb/elbUnhealthyInstances.js:65]:
```suggestion
                    helpers.addResult(results, 0, 'ELB does not have unhealthy instance', region, elbArn);
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]
Fix the failing test

**Review by mehakseedat63 — APPROVED** [review]

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**